contracts/utils/cryptography/P256.sol

```diff
@@ -183,6 +183,9 @@ library P256 {
     /**
      * @dev Point addition on the jacobian coordinates
      * Reference: https://www.hyperelliptic.org/EFD/g1p/auto-shortw-jacobian.html#addition-add-1998-cmo-2
+     *
+     * Note that if both input points are identical, this function will return `(0, 0, 0)` instead of doubling the point.
+     * The probability of identical input points happening naturally is negligible (approximately 2048 / (2^256 - 1), or about 10^-74).
```

Comment:
I'm not sure what is "natural" here

```suggestion
     * The probability of two random points being identical is negligible (approximately 2048 / (2^256 - 1), or about 10^-74).
```

Reply:
Also, how was that computed ? 

At first glace, it feals like since field of size P, so that probability of two values being identical is 1/P = 8.6e-78

Reply:
is not only 1 key out of P, as there are a few possible cases, from audit info:

there's two scenarios that lead to `_jAdd`:
1. `_preComputeJacobianPoints` calls `_jAdd`
2. `_jMultShamir` calls `_jAdd`

For the first scenario, there are 7 distinct valid weak private keys (`d`) that would trigger the bug: `d`∈{1,2,3,2<sup>−1</sup>,3<sup>−1</sup>,2 3<sup>−1</sup>,3 2<sup>−1</sup>}. Assuming that `d` is chosen uniformly at random among n<sup>−1</sup> values, where `n` is a 256<sup>256</sup> bit prime, the probability to choose any one of the weak keys is: 7/(2<sup>256</sup>-1).

For the second scenario, at most 2048 valid weak private keys would trigger the bug. Specifically, the probability of randomly drawing one of the weak keys is at most: 2048/(2<sup>256</sup>−1). Used this probability in the comment as it's more likely.